### PR TITLE
libsq wal integration

### PR DESCRIPTION
**Pull request by MarinPostma**
Basic integration of libsql-wal in sqld. Only the primary is supported right now, but it's still compatible with replicas with vanilla wal. Bottomless is not supported.

There are some tests failing with libsql-wal still (timeout issues in the simulation), i'll fix in a followup.

### CODE REVIEW

File: libsql-ffi/build.rs [hunk 1]
```diff
@@ -363,8 +363,8 @@ fn build_multiple_ciphers(out_path: &Path) {
 
     let mut make = Command::new("cmake");
     make.current_dir(sqlite3mc_build_dir.clone());
-    make.args(&["--build", "."]);
-    make.args(&["--config", "Release"]);
+    make.args(["--build", "."]);
+    make.args(["--config", "Release"]);
```
Comment (MarinPostma) [inline libsql-ffi/build.rs:367]: clippy stuff

File: libsql-ffi/build.rs [hunk 2]
```diff
@@ -303,7 +303,7 @@ fn build_multiple_ciphers(out_path: &Path) {
     let cxx = env("CXX");
 
     let toolchain_path = sqlite3mc_build_dir.join("toolchain.cmake");
-    let cmake_toolchain_opt = format!("-DCMAKE_TOOLCHAIN_FILE=toolchain.cmake");
+    let cmake_toolchain_opt = "-DCMAKE_TOOLCHAIN_FILE=toolchain.cmake".to_string();
```
Comment (MarinPostma) [inline libsql-ffi/build.rs:306]: clippy stuff

File: libsql-sys/src/wal/wrapper.rs
Comment (MarinPostma) [inline libsql-sys/src/wal/wrapper.rs:1]: clippy stuff

File: libsql-wal/src/name.rs [hunk 3]
```diff
@@ -3,7 +3,7 @@ use std::fmt;
 use bytes::Bytes;
 
 #[derive(Clone, PartialEq, Eq, Hash)]
-pub struct NamespaceName(Bytes);
+pub struct NamespaceName(pub Bytes);
```
Comment (LucioFranco) [inline libsql-wal/src/name.rs:6]: just expose a `from<Bytes>` rather than make it pub?

**Review by LucioFranco — APPROVED** [review]
Conditional approval

File: libsql-wal/src/name.rs (hunk 3 above)
Comment (MarinPostma) [reply libsql-wal/src/name.rs:6]: it's a placeholder type, it's going away asap

Comment (MarinPostma) [reply libsql-wal/src/name.rs:6]: I'm thinking I'll push the namespace type to the wal crate, and consume it in libsql-server. This will be a followup

**Pull request merged**